File: to-do-api/src/main/java/com/viet/to_do_api/dto/task/TaskDto.java
package com.viet.to_do_api.dto.task;

import java.time.LocalDateTime;
import java.util.List;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Schema(description = "Data transfer object for Task")
public class TaskDto {
    private Integer id;

    @NotBlank(message = "Title is required")
    private String title;

    private String description;

    private CategoryDto category;

    private List<TagDto> tags;

    private StatusDto currentStatus;

    private List<StatusDto> statusHistory;

    private List<CommentDto> comments;

    private List<AttachmentDto> attachments;

    @Min(value = 1, message = "Priority must be at least 1")
    @Max(value = 5, message = "Priority must be at most 5")
    private Integer priority;

    private LocalDateTime createdDate;

    private LocalDateTime modifiedDate;

    private LocalDateTime dueDate;

}

File: to-do-ui/src/app/models/task/task.dto.ts
import { CategoryDTO } from "./category.dto";
import { CommentDTO } from "./comment.dto";
import { StatusDTO } from "./status.dto";
import { TagDTO } from "./tag.dto";

export interface TaskDTO {
    id?: number;
    title?: string;
    description?: string;
    category?: CategoryDTO;
    tags?: TagDTO[];
    currentStatus?: StatusDTO;
    statusHistory?: StatusDTO[];
    comments?: CommentDTO[];
    attachments?: string[];
    priority?: number;
    createdDate?: string;
    modifiedDate?: string;
    dueDate?: string;
}

//  "id": 0,
//   "title": "",
//   "description": "",
//   "category": {},
//   "tags": [],
//   "currentStatus": {},
//   "statusHistory": [],
//   "comments": [],
//   "attachments": [],
//   "priority": 0,
//   "createdDate": "",
//   "modifiedDate": "",
//   "dueDate": ""